### ISSUE TITLE
Uncaught SyntaxError: Unexpected token ':'

### ISSUE THREAD

**Issue by loynoir** [sinclairzx81/typebox#356]
### Reproduce
```sh
$ grep typebox package.json 
    "@sinclair/typebox": "^0.26.4"
$ node
Welcome to Node.js v18.15.0.
Type ".help" for more information.
> var {Type} = await import('@sinclair/typebox'); var {TypeCompiler} = await import('@sinclair/typebox/compiler')
undefined
> void TypeCompiler.Compile(Type.Object({'node-mirror:release':Type.Optional(Type.String())}))
Uncaught SyntaxError: Unexpected token ':'
> 
```

```sh
$ deno
Deno 1.32.1
exit using ctrl+d, ctrl+c, or close()
REPL is running with all permissions allowed.
To specify permissions, run `deno repl` with allow flags.
> import { Type } from 'npm:@sinclair/typebox@0.26.4'; import { TypeCompiler } from 'npm:@sinclair/typebox@0.26.4/compiler'
undefined
> void TypeCompiler.Compile(Type.Object({'node-mirror:release':Type.Optional(Type.String())}))
Uncaught SyntaxError: Unexpected token ':'
    at Proxy.Function (<anonymous>)
    at Object.Compile (file:///home/vscode/.cache/deno/npm/registry.npmjs.org/@sinclair/typebox/0.26.4/compiler/compiler.js:548:45)
    at <anonymous>:2:19
```

### Related
https://github.com/sinclairzx81/typebox/issues/353

**Comment by sinclairzx81** [comment]
@loynoir Hi, that's embarrassing :D

The member expression encode was implemented on `0.26.4` (which was a bug), but somehow I missed the optional property check (which I note you had actually shown on your original issue code snippet). Sorry about that.

Have quickly published the additional expression encode for optional property checks on `0.26.5`. Will leave this issue open this time for you to close once you've had a chance to verify the fix. Again, thanks for reporting!

Cheers
S

**Comment by loynoir** [comment]
Checked with both `node` and `deno`, bug is gone.
